PCbuild/regen.vcxproj

```diff
@@ -0,0 +1,189 @@
+    <None Include="..\Python\graminit.c">
+    </None>
+    <None Include="..\Include\token.h">
+    </None>
+    <None Include="..\Include\opcode.h">
+    </None>
+    <None Include="..\Include\Python-ast.h">
+    </None>
+    <None Include="..\Python\Python-ast.c">
+    </None>
+    <None Include="..\Lib\keyword.py">
+    </None>
+    <None Include="..\Lib\symbol.py">
+    </None>
+    <None Include="..\Doc\library\token-list.inc">
+    </None>
+  </ItemGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+  <ImportGroup Label="ExtensionTargets">
+  </ImportGroup>
+  <Target Name="_RegenGrammar" BeforeTargets="Build">
+    <!-- Regenerate Include/graminit.h and Python/graminit.c from Grammar/Grammar using pgen-->
+    <Exec Command="&quot;$(PythonExe)&quot; -m Parser.pgen &quot;..\Grammar\Grammar&quot; &quot;..\Grammar\Tokens&quot; &quot;$(IntDir)graminit.h&quot; &quot;$(IntDir)graminit.c&quot;" />
+    <Copy SourceFiles="$(IntDir)graminit.h" DestinationFiles="..\Include\graminit.h">
+      <Output TaskParameter="CopiedFiles" ItemName="_UpdatedH" />
+    </Copy>
+    <Copy SourceFiles="$(IntDir)graminit.c" DestinationFiles="..\Python\graminit.c">
+      <Output TaskParameter="CopiedFiles" ItemName="_UpdatedC" />
+    </Copy>
+    <Warning Text="Grammar updated. You will need to rebuild pythoncore to see the changes." Condition="'@(_UpdatedH)' != '' and '@(_UpdatedC)' != ''" />
+  </Target>
+  <Target Name="_RegenAST_H" AfterTargets="_RegenGrammar">
+    <!-- Regenerate Include/Python-ast.h using Parser/asdl_c.py -h -->
+    <Exec Command="&quot;$(PythonExe)&quot; &quot;..\Parser\asdl_c.py&quot; -h &quot;$(IntDir)Python-ast.h&quot; &quot;..\Parser\Python.asdl&quot;" />
+    <Copy SourceFiles="$(IntDir)Python-ast.h" DestinationFiles="..\Include\Python-ast.h">
+      <Output TaskParameter="CopiedFiles" ItemName="_UpdatedH" />
+    </Copy>
+    <Warning Text="Python-ast.h updated. You will need to rebuild pythoncore to see the changes." Condition="'@(_UpdatedH)' != '' and '@(_UpdatedC)' != ''" />
+  </Target>
+  <Target Name="_RegenAST_C" AfterTargets="_RegenGrammar">
+    <!-- Regenerate Python/Python-ast.c using Parser/asdl_c.py -c -->
+    <Exec Command="&quot;$(PythonExe)&quot; &quot;..\Parser\asdl_c.py&quot; -c &quot;$(IntDir)Python-ast.c&quot; &quot;..\Parser\Python.asdl&quot;" />
+    <Copy SourceFiles="$(IntDir)Python-ast.c" DestinationFiles="..\Python\Python-ast.c">
+      <Output TaskParameter="CopiedFiles" ItemName="_UpdatedH" />
+    </Copy>
+    <Warning Text="Python-ast.c updated. You will need to rebuild pythoncore to see the changes." Condition="'@(_UpdatedH)' != '' and '@(_UpdatedC)' != ''" />
+  </Target>
+  <Target Name="_RegenOpcodes" AfterTargets="_RegenAST_C">
+    <!-- Regenerate Include/opcode.h from Lib/opcode.py using Tools/scripts/generate_opcode_h.py-->
+    <Exec Command="&quot;$(PythonExe)&quot; ..\Tools\scripts\generate_opcode_h.py &quot;..\Lib\opcode.py&quot; &quot;$(IntDir)opcode.h&quot;" />
```

Comment:
I'm not keen on the relative paths here. We have `$(PySourcePath)` for the root of the sources.